armi/bookkeeping/db/database3.py

```diff
@@ -100,6 +94,168 @@ def getH5GroupName(cycle: int, timeNode: int, statePointName: str = None) -> str
     return "c{:0>2}n{:0>2}{}".format(cycle, timeNode, statePointName or "")
 
 
+class JaggedArray:
+    def __init__(self, jaggedData, paramName):
+        """
+        Take a list of numpy arrays or lists and flatten them into a single 1D array.
+
+        This implementation can preserve the structure of a multi-dimensional numpy array
+        by storing the dimensions in self.shapes and then re-populating a numpy array of
+        that shape from the flattened 1D array. However, it can only preserve one layer of
+        jaggedness in a list of lists (or other iterables). For example, a list of tuples
+        with varying lengths can be flattened and reconstituted exactly. But, if a list of
+        lists of tuples is passed in, the tuples in that final layer of nesting will all be
+        flattened to a single 1D numpy array after a round trip. No structure is retained
+        from nested lists of lists or lists of tuples.
+
+        Parameters
+        ----------
+        jaggedData: list of numpy.ndarray
+            A list of numpy arrays (or lists) to be flattened into a single array
+        paramName: str
+            The name of the parameter represented by this data
+        """
+        offset = 0
+        flattenedArray = []
+        offsets = []
+        shapes = []
+        for array in jaggedData:
```

Comment:
MINOR thing: "array" is a bad variable name in Python because it conflicts with the Python standard library "import array".

You're not the first person to hit this annoyance. Sorry!